### PR TITLE
WIP: kube-proxy: remove OS conditionals.

### LINKED ISSUE #36277: Refactor kube-proxy to remove os conditional checks
`cmd/kube-proxy/app/server.go` has few conditional checks based on underlying OS like `runtime.GOOS == "windows"`, refactor `server.go` to do it more abstractly by moving them to `_linux/_windows` files.

All the conditional checks are in `NewProxyServerDefault()` and `Run()` methods and one solution could be to move them to `_linux/_windows` files

### PR DESCRIPTION
**Pull request by pires**
**What this PR does / why we need it**: in 1.5 we added Windows support to kube-proxy. This is part of an ongoing effort to improve what has been done. In this PR, we focused on removing OS checks based on [previous feedback](https://github.com/kubernetes/kubernetes/pull/36079#pullrequestreview-7272053).

**Which issue this PR fixes**: fixes #36277

**Special notes for your reviewer**: @thockin a second step we'd like to discuss is how can we remove the need for the Windows code to not know about `iptables.Interface` type. A third step could be to extract proxy modes as _plug-ins_, so that we can provide different implementations for Windows as well.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-proxy/app/server_unsupported.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, error) {
+	return nil, fmt.Errorf("Proxying is not supported in this build")
+}
+
+// Tune conntrack, if requested
+func (s *ProxyServer) tuneConnTracker() error {
+	return fmt.Errorf("Proxying is not supported in this build")
+}
+
+func getConntrackMax(config *options.ProxyServerConfig) (int, error) {
```
Comment (thockin) [inline cmd/kube-proxy/app/server_unsupported.go]: why is this part of the "interface" only used in Linux.

File: cmd/kube-proxy/app/server_unsupported.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"k8s.io/kubernetes/cmd/kube-proxy/app/options"
+)
+
+// NewProxyServerDefault creates a new ProxyServer object with default parameters.
+func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, error) {
+	return nil, fmt.Errorf("Proxying is not supported in this build")
+}
+
+// Tune conntrack, if requested
+func (s *ProxyServer) tuneConnTracker() error {
```
Comment (thockin) [inline cmd/kube-proxy/app/server_unsupported.go]: why isn't this just embedded in the _linux code?

**Review by thockin — COMMENTED** [review]
This feels partial.  The userspace and iptables modes only apply on Linux, so why are they in the os-agnostic section?

File: cmd/kube-proxy/app/server_unsupported.go (hunk 2 above)
Comment (pires) [inline cmd/kube-proxy/app/server_unsupported.go]: conntrack is something we can't do in Windows, yet. We're evaluating new solutions like OpenVSwitch to deliver this and much more, e.g. stateful firewalling (flows). We're just trying to _interface_ it as much as we can right now so it becomes easier in the future to implement for Windows.

File: cmd/kube-proxy/app/server_unsupported.go (hunk 1 above)
Comment (pires) [inline cmd/kube-proxy/app/server_unsupported.go]: Same comment as above.

**Comment by pires** [comment]
@jbhurat I have rebased the PR so make sure you have updated your working copy.

File: cmd/kube-proxy/app/server_unsupported.go (hunk 1 above)
Comment (pires) [inline cmd/kube-proxy/app/server_unsupported.go]: Correct.

File: cmd/kube-proxy/app/server_unsupported.go (hunk 2 above)
Comment (pires) [inline cmd/kube-proxy/app/server_unsupported.go]: We're working on having connection tracking on Windows. Even though, I have reviewed partially addressed your concerns in a new commit.

**Comment by pires** [comment]
@k8s-bot cross build this

**Comment by pires** [comment]
@k8s-bot cross build this

**Comment by pires** [comment]
@k8s-bot gce etcd3 e2e test this

**Pull request closed without merging**